### ISSUE TITLE
Weekly community Check-In #30 - Modifying Expectations

### ISSUE THREAD

**Issue by cesswairimu** [publiclab/community-toolbox#237]
Hi everyone :wave: !

We all at Public Lab :balloon:  - learn, grow, work, brainstorm ideas, contribute together so why not share about our _weekly goals and the awesome work_ we have done at Public Lab with each other, so we can support and collaborate with each other better. We have a Community Check-In each week, where every community member can share something about their work from the past week and about their current week's goal :dart: . You are also welcome to share fun-fact :smile: , new ideas :bulb: , your learning goals :ballot_box_with_check: .

We believe in collaborative efforts to support our community. We are running a learning platform which helps a newcomer to become master of tomorrow. :100:

If you're new here, welcome, and please comment a _Hello_ below, we would love to work with you. If you're looking for new issues, please try some of our [first-timers-only](https://code.publiclab.org/#r=all) issues.

We're SO EXCITED to have your help!

**Is there anything, you would like to share with us from past week's work? What is your plan for this week?**

If you have not planned yet, just leave a Hello! :wave: so that we know that you are in sync with us :arrows_clockwise: and doing well!

As always, if you're waiting for a review, or if you're stuck, please request help here OR leave a comment with @publiclab/mentors @publiclab/reviewers for some input. :raised_hands:

#### Gitter

Gitter is an active chatroom in our community and we'll be sending weekly reminders about check-ins there. Be sure to sign up [there](https://gitter.im/publiclab/publiclab) for these updates or just to join the conversation. You can also join us through http://publiclab.org/chat :speech_balloon:.

**This Week's Theme:**

**Modifying Expectations**

We  all have an idea in mind on how we are going to implement an issue when we claim it, sometimes that does not go as planned. Lets talk about that..
The interns started the internship with a specific set of project goals. However, usually those goals need to be modified, and that's perfectly fine!

Delays to projects happen. Maybe your project turned out to be more complicated than you anticipated. Maybe you needed to learn some concepts before you could tackle project tasks. . These are all perfectly valid reasons for projects to be a bit behind schedule or divert to other tasks you did not have in mind when drafting the proposal.

This week lets share on how the project outline and timelines you had on your proposals, have been modified as you started working on your projects.

## :memo:  Note to Summer of Code interns:
We request all the SoC students to read and include necessary points in their check-in comment

- Update on what you worked on last week
- Goals in the coming week
- What things you did collaboratively last week? This is really important we want team spirit.
It is like a weekly scrum. Be flexible.
- GSOC interns 2nd eval due on 26th July and please share links to blogs if available.
- Outreachy interns there is "Career opportunities" chat on  23rd July  1p.m UTC on Zulip
- Feel free to tell us about your absence too, if you're taking a break.
- Issue/PR you're struggling with (if any)

Join us for Open Code Call/ Weekly scrum on  Tuesday - http://publiclab.org/opencall 

Thanks all and have a great week ahead :balloon: 

_Theme from Outreachy week 7_

**Comment by cesswairimu** [comment]
@milaaraujo @kevinzluo @Rishabh570 @sashadev-sky @shubhscoder @sidntrivedi012 @SidharthBansal @stefannibrasil @ViditChitkara @tech4GT @mridulnagpal @IshaGupta18 @CoderJolly @namangupta01 @sagarpreet-chadha @Souravirus @MayankKashyap @siaw23 @ryzokuken @icarito @steviepubliclab, @ebarry, @jywarren, @sagarpreet-chadha, @JonathanXu1 @rexagod, @divyabaid16, @dinaelhanan, @oorjitchowdhary @publiclab/mentors @publiclab/image-sequencer-guides @publiclab/leaflet-environmental-layers-guides @Paarmita @publiclab/reviewers @Divy123 @bhavayAnand9 @Mridul97 @avsingh999 @Dhiraj240 @codeIriss @IgorWilbert @pdurbin @HarshKhandeparkar @niravasher @publiclab/reviewers @publiclab/is-reviewers @publiclab/community-reps @divyabaid16 @publiclab/community-reps @subhahu123 @govindjeevan @gautamig54 @rarrunategu1 @CleverFool77 @ananyaarun @ananya @publiclab-mimi @ananya @hc-barker @starkblaze01 @aSquare14 @vaarigupta @alonpeer @pdurbin @HarshKhandeparkar @newbazz

**Comment by cesswairimu** [comment]
@newbazz @aSquare14 @vaarigupta @alonpeer @aashna27 @harshithpabbati @jesus-montano  @cryptoclidus @mohamed-elalem @konolga@gowtham5699

**Comment by MargaretAN9** [comment]
**This note is a GSOC Phase 2 summary and updates Phase 3 goals.**

**Image Sequencer Manual** Substantial changes to the manual were made over this rating period. Proposed future changes for the next period include: -Adding new modules such as Noise reduction, Minify Image, blob Analysis, colorimetry -Demonstration of Insert step functionality -Increasing the applications section to include blob detection and colorimetry -Adding an intro section which list modules according to their general functions; such as color modification, size or shape changes, text/QR,etc. (This approach may help develop submenus vs the current 40 long dropdown list) . Github pull at: https://github.com/publiclab/image-sequencer/pull/1171 Initial announcement: https://publiclab.org/notes/MaggPi/06-10-2019/draft-of-public-lab-image-sequencer-user-manual-comments-welcome Manual Version 0.3: https://github.com/MargaretAN9/GSOC-2019/blob/master/ImageSequencerUserManual.md#saturation-module

**IS/OpenCV** Demonstrated the ability to load opencv and then process images from datauri to string format and then to rgba. Have not been able to successfully transfer image from rgba back into IS datauri display format. At this point, aashna27 great work has superseded my efforts and it’s probably best to align with her designs. My initial code trying to develop IS/openCV in a single file is available at : https://github.com/MargaretAN9/GSOC-2019/tree/master/is%20w%20opencv Once opencv is pushed, several new options exist for IS modules. One possible option is a calibration module for microscope pictures which measure a known reference and provide data for the grid module. 
Additional work includes a post demonstrating VS Code as an operating environment for OpenCV.js: https://publiclab.org/notes/MaggPi/07-04-2019/introduction-to-image-processing-with-opencv-js

**Satellite Processing** Steady progress using python to load and display satellite images (Landsat or Planet). I was hoping to develop a request to only download a portion vs the entire image. Comments from Planet.com indicate that this is not possible. The current approach is to use API’s to select thumbnail areas of interest, download geoTIFF files and then use gdal to crop images for easier processing.

Currently working an application which uses satellite imagery to update OSM data with environmental characteristics. This Mapknitter link demonstrates how overpass-turbo can be used to identify park areas in satellite images, which then can then be processed to update OSM markings. Code available at: https://github.com/MargaretAN9/GSOC-2019/edit/master/sattelliteImageProcess/greenOperpassExV1.ipynb

Collaboration efforts include Q/A’s to anaconda, opencv, planet.com and IS github sites.

**Comment by jywarren** [comment]
Hi all! Great theme, Cess. Adjusting plans, incorporating learnings, and being comfortable with this, is an important part of our work! ❤️ 

Thanks to everyone who's been posting their GSoC updates on PublicLab.org. I'm catching up again after some travel and hope everyone is doing well!

**Comment by SidharthBansal** [comment]
Hello Everyone!!!
I came back after 10 week internship at Google India. Now I am in great zeal to join the community again. My internship ended today, and I am here. I love PL. It is the best open source organization.

I hope everyone is with great zeal for the 3rd eval of SoC Programs. Best of luck for the results.
Great theme Cess. 
Let's discuss about evolutionary software development life cycle model so that your requirements can fit into your design well and you can save your time at later phases of development. 
No system can be built in perfect way in the first run. Try to solve your query for basic prototype first. Basic prototype should deal with the necessary requirements. As you grow, add on functionality to your prototype. Once you think your prototype is ready. Don't go and implement it. Try to think of all the possibilities and edge cases. Now make changes in the design of the prototype according to them. Then implement it. 
Even after it's implementation, you will find certain cases which are left. Then directly try to embed those functionalities into your code.
And don't be afraid of failures which you are designing things. Things will break. Failure is a sign of trying. Giving up is wrong. Thomas invented bulb after 2000+ failures because he knew 2000+ ways not to make the bulb. I am giving you this example, because Software Engineers see a lot of bugs/errors and many things which frustrate us and are not up to our expectation. We should not lose hope.
If you will set lower expectation from yourself, then you will attain it. After attaining it you will feel positive and will have hope. If you will have higher expectations, then you are likely to fail.
Gradually increase the expectation as you rise. But always keep attainable expectations.


My goal of this week is to relax and see what all is going on at Public Lab. I have to catch all the amazing SoCers are doing here. 

Thanks all
Sidharth

**Comment by IshaGupta18** [comment]
Hey Everyone!
Working on simple-data-grapher

@jywarren @sagarpreet-chadha @gauravano please please review the integration PR at plots2 
https://github.com/publiclab/plots2/pull/5993. It has been waiting for quite a long time now!

I am almost done with UI testing here https://github.com/publiclab/simple-data-grapher/pull/73

I am happy to tell you that the "Publish as a research note" is 80% complete and I would be creating a PR for the same, I was waiting because I wanted the integration PR to get merged first. This one was I think the toughest feature of the project! https://github.com/publiclab/simple-data-grapher/issues/65

I have to write a couple of more tests and documentation and work on the UI, which is the final stage of the project!

Have a great week everyone!

**Comment by alaxalves** [comment]
My expectations have been modified for the better, by modifying my project and schedule(shortening it), as Igor Wilbert(one of my mentors) had suggested me and Kaustubh got to finish our planned work a little ahead of schedule - thanks to Sasha as well, she has been helping us sooo much.
This is giving us a chance to work out a little more along other PL projects. And this is great. 

**Update on what you worked on last week:**

Last week and the upcoming week I'll be very busy with college, going out for documentation and such, so I'll be a little of the radar. But I have kinda prep'd myself for this, so I had Kaustubh finishing Rails 6 upgrade and I worked along with Sasha to polish what was left on Rails 5 - in order to get it on main branch asap.

**Goals in the coming week**

I intend to help Vidit Chitkara with his Action Cable/Puma/Foreman setup, in order to get it properly running on top of Rails 5. We've talked and have some ideas for it.
Other goal is to help Kaustubh in setting system tests in MK, the same setup as we have in Plots2. Besides those I intend to finally "polish" what has been left in Rails 6 in order to get it on main branch too.

**What things you did collaboratively last week? This is really important we want team spirit.**

Me and Sasha have identified some issues in Rails 5 regarding the Map Loading and Map creation, but turned out it was an issue in main branch. We have fixed both, and waiting on the merge to happen.
